<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ImageTaskRunnerWidget</class>
 <widget class="QWidget" name="ImageTaskRunnerWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>299</width>
    <height>414</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="MinimumExpanding" vsizetype="Preferred">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string notr="true"/>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <layout class="QVBoxLayout" name="optionsLayout">
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_15">
       <item>
        <widget class="QLabel" name="candidateShapesPerStepLabel">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="text">
          <string extracomment="Label text next to a slider that allows the user to control the number of geometric shapes the app creates each time it updates">Shapes Per Update</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSlider" name="candidateShapesPerStepSlider">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimum">
          <number>1</number>
         </property>
         <property name="maximum">
          <number>1500</number>
         </property>
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="candidateShapesPerStepCountLabel">
         <property name="text">
          <string notr="true">0</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_16">
       <item>
        <widget class="QLabel" name="mutationsPerCandidateShapeLabel">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="text">
          <string extracomment="Label text next to a slider that allows the user to control the maximum number of times a shape is modified (mutated) each time it updates">Max Shape Mutations</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSlider" name="mutationsPerCandidateShapeSlider">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimum">
          <number>1</number>
         </property>
         <property name="maximum">
          <number>1500</number>
         </property>
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="mutationsPerCandidateShapeCountLabel">
         <property name="text">
          <string notr="true">0</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_3">
       <item>
        <widget class="QLabel" name="shapeOpacityLabel">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="text">
          <string extracomment="Label text next to a slider that allows the user to control opacity/transparency of the geometric shapes that are created">Shape Opacity</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSlider" name="shapeOpacitySlider">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimum">
          <number>1</number>
         </property>
         <property name="maximum">
          <number>255</number>
         </property>
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="tickPosition">
          <enum>QSlider::NoTicks</enum>
         </property>
         <property name="tickInterval">
          <number>5</number>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="shapeOpacityValueLabel">
         <property name="text">
          <string notr="true">0</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QFormLayout" name="formLayout">
       <item row="0" column="0">
        <widget class="QLabel" name="randomSeedLabel">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="text">
          <string extracomment="Label text next to the seed value of a random number generator. The seed number is a mathematical term - not like the seed of a plant!">Random Seed</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QSpinBox" name="randomSeedSpinBox">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="alignment">
          <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
         </property>
         <property name="correctionMode">
          <enum>QAbstractSpinBox::CorrectToNearestValue</enum>
         </property>
         <property name="showGroupSeparator" stdset="0">
          <bool>false</bool>
         </property>
         <property name="maximum">
          <number>2147483647</number>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QFormLayout" name="formLayout_2">
       <item row="0" column="0">
        <widget class="QLabel" name="maxThreadsLabel">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="text">
          <string extracomment="A text label next to a value that describes the maximum number of CPU threads the computer can use when recreating an image using shapes">Max Threads</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QSpinBox" name="maxThreadsSpinBox">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="alignment">
          <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
         </property>
         <property name="correctionMode">
          <enum>QAbstractSpinBox::CorrectToNearestValue</enum>
         </property>
         <property name="maximum">
          <number>32</number>
         </property>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </item>
   <item>
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeType">
      <enum>QSizePolicy::Fixed</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>10</height>
      </size>
     </property>
    </spacer>
   </item>
   <item>
    <widget class="Line" name="line">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QWidget" name="shapeTypesContainer" native="true">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <widget class="QCheckBox" name="usesRotatedRectangles">
      <property name="geometry">
       <rect>
        <x>6</x>
        <y>26</y>
        <width>88</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string extracomment="Plural form of rectangle geometric shape. These rectangles are rotated through an angle (0-360 degrees)">Rotated Rectangles</string>
      </property>
     </widget>
     <widget class="QCheckBox" name="usesCircles">
      <property name="geometry">
       <rect>
        <x>6</x>
        <y>6</y>
        <width>42</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string extracomment="Plural form of circle/circular geometric shape">Circles</string>
      </property>
     </widget>
     <widget class="QCheckBox" name="usesQuadraticBeziers">
      <property name="geometry">
       <rect>
        <x>6</x>
        <y>166</y>
        <width>81</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string extracomment="Plural form of quadratic bezier curve geometric shape">Quadratic Beziers</string>
      </property>
     </widget>
     <widget class="QCheckBox" name="usesEllipses">
      <property name="geometry">
       <rect>
        <x>6</x>
        <y>146</y>
        <width>45</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string extracomment="Plural form of ellipse/ellipsoid geometric shape">Ellipses</string>
      </property>
     </widget>
     <widget class="QCheckBox" name="usesTriangles">
      <property name="geometry">
       <rect>
        <x>6</x>
        <y>126</y>
        <width>52</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string extracomment="Plural form of triangle geometric shape">Triangles</string>
      </property>
     </widget>
     <widget class="QCheckBox" name="usesRectangles">
      <property name="geometry">
       <rect>
        <x>6</x>
        <y>106</y>
        <width>58</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string extracomment="Plural form of rectangle geometric shape">Rectangles</string>
      </property>
     </widget>
     <widget class="QCheckBox" name="usesLines">
      <property name="geometry">
       <rect>
        <x>6</x>
        <y>86</y>
        <width>38</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string extracomment="Plural form of the simple line geometric shape">Lines</string>
      </property>
     </widget>
     <widget class="QCheckBox" name="usesRotatedEllipses">
      <property name="geometry">
       <rect>
        <x>6</x>
        <y>66</y>
        <width>75</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string extracomment="Plural form of ellipse geometric shape. These ellipses are rotated through an angle (0-360 degrees)">Rotated Ellipses</string>
      </property>
     </widget>
     <widget class="QCheckBox" name="usesPolylines">
      <property name="geometry">
       <rect>
        <x>6</x>
        <y>46</y>
        <width>50</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string extracomment="Plural form of the polyline geometric shape (multiple line segments joined together)">Polylines</string>
      </property>
     </widget>
    </widget>
   </item>
   <item>
    <widget class="Line" name="line_3">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QPushButton" name="runStopButton">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="text">
      <string notr="true">Run</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QPushButton" name="stepButton">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="text">
      <string extracomment="Text on a button that updates (steps) a task once when pressed">Step</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QPushButton" name="clearButton">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="text">
      <string extracomment="Text on a button that clears/resets the results of a task back to a starting value when pressed">Clear</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QWidget" name="bottomButtonsContainer" native="true"/>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>
